Src/BootCamp.Chapter/BalanceStats.cs

```diff
@@ -1,37 +1,294 @@
-﻿namespace BootCamp.Chapter
+﻿using System;
+using System.Globalization;
+using System.Linq;
+using System.Text;
+
+namespace BootCamp.Chapter
 {
     public static class BalanceStats
     {
+        private const string InValidOutput = "N/A.";
+
+        public static bool IsInValidValidInput(string[] input) => input == null || input.Length == 0;
+
         /// <summary>
         /// Return name and balance(current) of person who had the biggest historic balance.
         /// </summary>
         public static string FindHighestBalanceEver(string[] peopleAndBalances)
         {
-            return "";
+            if (IsInValidValidInput(peopleAndBalances))
+            {
+                return InValidOutput;
+            }
+
+            var personWithHighestBalance = new StringBuilder();
+            var highestBalanceEver = FindHighestBalance(peopleAndBalances, personWithHighestBalance);
+
+            if (personWithHighestBalance.ToString().Contains(", "))
+            {
+                personWithHighestBalance = ReplaceCommaWithAnd(personWithHighestBalance); 
+            }
+
+            return $"{personWithHighestBalance.ToString()} had the most money ever. ¤{highestBalanceEver}.";
+        }
+
+        private static StringBuilder ReplaceCommaWithAnd(StringBuilder message)
+        {
+            var index = message.ToString().LastIndexOf(", ");
+            return message.Remove(index, 2).Insert(index, " and "); 
+        }
+
+        private static decimal FindHighestBalance(string[] peopleAndBalances, StringBuilder PersonWithHighestBalance)
+        {
+            var HighestBalanceEver = decimal.MinValue;
+
+            for (int i = 0; i <= peopleAndBalances.Length - 1; i++)
+            {
+                var currentPersonData = peopleAndBalances[i].Split(',');
+                try
+                {
+                    var highestAmountOfPerson = decimal.Parse(currentPersonData[1..].Max());
+                    
+                    
+                    if (highestAmountOfPerson == HighestBalanceEver)
+                    {
+                        PersonWithHighestBalance.Append(", ");
+                        PersonWithHighestBalance.Append(currentPersonData[0]);
+                    }
+
+                    if (highestAmountOfPerson > HighestBalanceEver)
+                    {
+                        HighestBalanceEver = highestAmountOfPerson;
+                        var currentOne = PersonWithHighestBalance.ToString();
+
+                        if (string.IsNullOrEmpty(currentOne))
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Refactor this code to not nest more than 3 control flow statements.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4967975)

Reply:
done